src/catalog-debug/src/main.rs

```diff
@@ -610,7 +610,12 @@ async fn upgrade_check(
                 secrets_reader,
                 None,
             ),
-            builtin_item_migration_config: BuiltinItemMigrationConfig::Legacy,
+            builtin_item_migration_config: BuiltinItemMigrationConfig {
+                // We don't actually want to write anything down, so use an in-memory persist
+                // client.
+                persist_client: PersistClient::new_for_tests().await,
```

Comment:
This is the only scariest part of the PR IMO since it's hard (impossible?) to test the upgrade checks in the cloud repo.